File: lib_openweather/src/main/java/guepardoapps/library/openweather/utils/BitmapHelper.java
package guepardoapps.library.openweather.utils;

import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.PorterDuff;
import android.graphics.PorterDuffXfermode;
import android.graphics.Rect;
import android.graphics.RectF;
import android.support.annotation.NonNull;

@SuppressWarnings({"WeakerAccess"})
public class BitmapHelper {

    /**
     * @param bitmap the original bitmap
     * @param height height of the bitmap to return
     * @param width  width of the bitmap to return
     * @return returns a rounded bitmap
     */
    public static Bitmap GetCircleBitmap(@NonNull Bitmap bitmap, int height, int width) {
        final Bitmap output = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
        final Canvas canvas = new Canvas(output);

        final int color = Color.BLACK;
        final Paint paint = new Paint();
        final Rect rect = new Rect(0, 0, width, height);
        final RectF rectF = new RectF(rect);

        paint.setAntiAlias(true);
        canvas.drawARGB(0, 0, 0, 0);
        paint.setColor(color);
        canvas.drawOval(rectF, paint);

        paint.setXfermode(new PorterDuffXfermode(PorterDuff.Mode.SRC_IN));
        canvas.drawBitmap(bitmap, rect, rect, paint);

        bitmap.recycle();

        return output;
    }
}

File: app/src/main/java/guepardoapps/lib/openweather/utils/BitmapHelper.kt
package guepardoapps.lib.openweather.utils

import android.graphics.*
import android.support.annotation.NonNull

class BitmapHelper {
    companion object {
        fun getCircleBitmap(@NonNull bitmap: Bitmap,
                            height: Int,
                            width: Int,
                            color: Int): Bitmap {
            val output: Bitmap = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888)

            val rect = Rect(0, 0, width, height)
            val rectF = RectF(rect)

            val paint = Paint()
            paint.isAntiAlias = true
            paint.color = color
            paint.xfermode = PorterDuffXfermode(PorterDuff.Mode.SRC_IN)

            val canvas = Canvas(output)
            canvas.drawARGB(0, 0, 0, 0)
            canvas.drawOval(rectF, paint)
            canvas.drawBitmap(bitmap, rect, rect, paint)

            bitmap.recycle()

            return output
        }
    }
}